kernel/src/schema/mod.rs

```diff
@@ -458,6 +458,73 @@ impl StructType {
     }
 }
 
+impl IntoIterator for StructType {
+    type Item = StructField;
+    type IntoIter = StructFieldIntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StructFieldIntoIter {
+            inner: self.fields.into_values(),
+        }
+    }
+}
+
+impl<'a> IntoIterator for &'a StructType {
+    type Item = &'a StructField;
+    type IntoIter = StructFieldRefIter<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StructFieldRefIter {
+            inner: self.fields.values(),
+        }
+    }
+}
+
+pub struct StructFieldIntoIter {
+    inner: indexmap::map::IntoValues<String, StructField>,
+}
+
+impl Iterator for StructFieldIntoIter {
+    type Item = StructField;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.next()
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.inner.size_hint()
+    }
```

Comment:
can definitely take this in a follow up but wanted to raise that it's probably very beneficial to override a handful of performance-sensitive operations that can go from O(N) to O(1) if we pass through to `inner`. consider
- `count()` --> I think this is easy
- `last()` --> I think easy since it's double-ended
- `nth()`  --> I think inner should have `nth()`?